recipes/cpp_project_framework/all/test_package/CMakeLists.txt

```
@@ -0,0 +1,17 @@
+cmake_minimum_required(VERSION 3.1)
+project(PackageTest CXX)
+
+set(CMAKE_CXX_STANDARD 14)
+set(CMAKE_CXX_STANDARD_REQUIRED True)
+
+include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+conan_basic_setup()
+
+find_package(GTest REQUIRED)
+
+add_executable(test_package test_package.cpp)
+if(UNIX)
+	target_link_libraries(test_package GTest::gtest GTest::gtest_main stdc++fs)
+else()
+	target_link_libraries(test_package GTest::gtest GTest::gtest_main)
```

```suggestion
    target_link_libraries(test_package cpp_project_framework::cpp_project_framework GTest::gtest GTest::gtest_main stdc++fs)
else()
    target_link_libraries(test_package cpp_project_framework::cpp_project_framework GTest::gtest GTest::gtest_main)
```
Find the target is the first step, you also need to add as dependency, otherwise, cmake will not include. That's why it didn't find your headers before.

---

changed in latest version.

---

I changed according to your suggestion.  However, I am not in favor of this approach.  I would like to use Conan to manage my dependencies but somehow this approach requires me to add the dependencies all by myself in CMake file.  The most weird thing is that I need to "link" to a header-only library.

---

Conan aims to be transparent, it provides the same information and uses the same mechanism as if you did `cmake --install`

If gtest was a requirement (and not an add on for the test) you would not need to add it here.

It does manage your dependencies and provides exactly like CMake would have done

---

Thanks~  I will try to understand the philosophy behind.